src/components/projecteditor/control/backend.js

```diff
@@ -692,19 +729,60 @@ export default class Backend {
         setTimeout(() => cb(0, projects), 1);
     };
 
-    createProject = (files, cb) => {
+    // get Files belonging to a given inode
+    getProjectFiles = (inode, cb) => {
+        const data =
+            JSON.parse(localStorage.getItem(DAPP_FORMAT_VERSION)) || {};
+        if (!data.projects) data.projects = [];
+
+        const project = data.projects.filter(item => {
+            return item.inode === inode;
+        });
+
+        if(project.length){
+            // project is not empty
+            cb(project[0].files);
+        } else {
+            console.error("Project with given inode doesn't exist");
+        }
+
+    };
+
+    createProject = (files, cb, isTemporary) => {
         const data =
             JSON.parse(localStorage.getItem(DAPP_FORMAT_VERSION)) || {};
 
         if (!data.projects) data.projects = [];
 
-        const inode = Math.floor(Math.random() * 10000000);
-        // TODO: check if project with this inode already exists.
-        const project = {
+        let inode = 1;
+
+        if (!isTemporary) {
+            // smallest random number to be generated is 2, 1 is reserved
+            inode = this.generateRandomInode();
+        }
+
+        let project = {
             inode: inode,
             files: files,
         };
-        data.projects.push(project);
+
+        // check if project with given inode already exists.
+         if (this.isDuplicateInode(data.projects, inode)) {
+
+             if (isTemporary) {
+                 // if duplicate with inode == 1, overwrite
+                 data.projects = data.projects.filter(function( project ) {
+                     return project.inode !== inode;
+                 });
+                 data.projects.push(project);
+             } else {
+                 // generate new inode
+                 project.inode = this.generateRandomInode();
```

Comment:
This logic does not guarantee an unique inode, since it only checks for a collision the first time.

Reply:
@Bashlund fixed in the latest commit

Reply:
It still has dead code in the `createProject` function.

Reply:
Something like this :)
```
        let inode = 1;

        if (isTemporary) {
            if (this.isDuplicateInode(data.projects, inode)) {
                // if duplicate with inode == 1, overwrite
                data.projects = data.projects.filter(function( project ) {
                    return project.inode !== inode;
                });
            }
        }
        else {
            // smallest random number to be generated is 2, 1 is reserved for temporary projects
            inode = this.generateRandomInode(data.projects);
        }

        let project = {
            inode: inode,
            files: files,
        };

        data.projects.push(project);
```

Reply:
me gusta :)